pkg/utils/chart.go

```diff
@@ -328,3 +342,38 @@ func getReleaseName(hr *appsapi.HelmRelease) string {
 	}
 	return releaseName
 }
+
+func replaceCRDs(cfg *action.Configuration, hr *appsapi.HelmRelease, chart *chart.Chart) error {
+	if hr.Spec.SkipCRDs != nil && *hr.Spec.SkipCRDs {
+		return nil
+	}
+
+	if hr.Spec.ReplaceCRDs != nil && *hr.Spec.ReplaceCRDs {
+		if errs := doReplaceCRDs(cfg, chart); len(errs) > 0 {
+			return errors.NewAggregate(errs)
+		}
+	}
+	return nil
+}
+
+func doReplaceCRDs(cfg *action.Configuration, targetChart *chart.Chart) []error {
```

Comment:
This should be `error`, instead of `[]error`. Using `error` will keep all the methods with a consistent style. And we don't need to check the error length.

Use `errors.NewAggregate(errs)`.

Reply:
I understand, but doReplaceCRDs call itself recursively, if `error` used, I have to use type assertion and collect errors as following, do you have better ways?
```
err := doReplaceCRDs(cfg, dep)
allErrs = append(allErrs, err.(errors.Aggregate).Errors()...)
```

Reply:
```go
if err := doReplaceCRDs(cfg, dep); err != nil {
    allErrs = append(allErrs, err)
}
```